### PR TITLE
chore: add github ci

### LINKED ISSUE #20: Add CI
Github Actions Workflow similar to [this](https://github.com/numaproj/numaflow/blob/main/.github/workflows/ci.yaml)

### PR DESCRIPTION
**Pull request by xdevxy**
Closes #20 

Adds CI with unit-test and lint as a starter.

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
+name: test
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+    branches:
+      - "main"
+      - "ci-test"
```
Comment (xdevxy) [inline .github/workflows/ci.yaml:9]: Put it there to have a branch for any ci testing.

Comment (juliev0) [reply .github/workflows/ci.yaml:9]: that's a neat idea - so if we change this file in the future, we should use this same branch to do it?

Comment (juliev0) [reply .github/workflows/ci.yaml:9]: Sorry, what's the case where we would want to merge into "ci-test" instead of "main"?

Comment (xdevxy) [reply .github/workflows/ci.yaml:9]: this is only to ensure the CI change got triggered on this branch to verify it is correct. If we don't want it there, I can do a follow up to remove it.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - name: Set up Go 1.x
+        uses: actions/setup-go@v4.0.0
+        with:
+          go-version: '1.20'
+        id: go
```
[the remark this answers is not in the document]
Comment (xdevxy) [reply .github/workflows/ci.yaml:20]: :+1  thanks for the suggestion. I see this is a nice optimization. If you don't see strong reason to not do it manually.  Let's do it in a follow up.

**Comment by xdevxy** [comment]
I think we should merge this as soon as possible, as it is already catching errors. See the lint failure.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (xdevxy) [reply .github/workflows/ci.yaml]: Sorry @juliev0  I confused you! :D I misused this setting which is meant for identifying branch `targeting` to not currently on. Thanks for pointing out.

**Review by juliev0 — APPROVED** [review]

**Pull request merged**